pkgs/applications/system/coolercontrol/coolercontrol-gui.nix

```diff
@@ -0,0 +1,66 @@
+{ python3
+, qtbase
+, qtsvg
+, qtimageformats
+, qtwayland
+, wrapQtAppsHook
+}:
+
+{ version
+, src
+, meta
+}:
+
+python3.pkgs.buildPythonApplication {
+  pname = "coolercontrol";
+  inherit version src;
+  sourceRoot = "source/coolercontrol-gui";
+  format = "pyproject";
+
+  pythonRelaxDeps = true;
+
+  pythonRemoveDeps = [ "pyside6" ]; # resolution issue with the wheel and relaxed dependencies
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ] ++ (with python3.pkgs; [
+    poetry-core
+    setuptools
+    pythonRelaxDepsHook
+  ]);
+
+  buildInputs = [
+    qtbase
+    qtimageformats
+    qtsvg
+    qtwayland
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyside6
+    apscheduler
+    matplotlib
+    numpy
+    setproctitle
+    jeepney
+    requests
+    dataclass-wizard
+  ];
+
+  postInstall = ''
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.desktop" -t "$out/share/applications/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.metainfo.xml" -t "$out/share/metainfo/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.png" -t "$out/share/icons/hicolor/256x256/apps/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.svg" -t "$out/share/icons/hicolor/scalable/apps/"
+  '';
+
+  dontWrapQtApps = true;
```

Comment:
Out of curiousity, why is this needed/preferred?

Reply:
There were 2 layers of wrappers with that binary: the Python wrapper and the Qt wrapper.

```
↪ env LANG=C ls -ahl /nix/store/54dx6wafi9850ci9kshs3n1s2sqr4mfg-coolercontrol-0.17.0/bin/
total 40K
dr-xr-xr-x 2 root root   96 Jan  1  1970 .
dr-xr-xr-x 6 root root   60 Jan  1  1970 ..
-r-xr-xr-x 1 root root 4.1K Jan  1  1970 ..coolercontrol-wrapped-wrapped
-r-xr-xr-x 1 root root  16K Jan  1  1970 .coolercontrol-wrapped
-r-xr-xr-x 1 root root  16K Jan  1  1970 coolercontrol
```

This just disables the automatic wrapping of the Qt one, and the `preFixup` line consolidates its `makeWrapper` flags into the options for the Python one. Manual entry for this: [When using `wrapGAppsHook` with special derivers you can end up with double wrapped binaries.](https://nixos.org/manual/nixpkgs/stable/#ssec-gnome-common-issues-double-wrapped)

Reply:
aha thanks!